### ISSUE TITLE
Heisenbug in `group-chat.spec.js`

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#2256]
### Problem

https://github.com/user-attachments/assets/772eebd1-dacc-44e3-95ee-8a99ed0f2a48

- Example [failed build](https://github.com/okTurtles/group-income/actions/runs/10101571423/job/27935432333?pr=2255)
- Example [Cypress Dasboard issue](https://cloud.cypress.io/projects/q6whky/runs/2803/specs?utm_source=github)

Test failure info:

```
    1) user3 joins the Dreamers group and general channel again and logout

  15 passing (2m)
  1 pending
  1 failing

  1) Group Chat Basic Features (Create & Join & Leave & Close)
       user3 joins the Dreamers group and general channel again and logout:
     AssertionError: Timed out retrying after 30000ms: expected '<span.c-twrapper>' to contain 'user3-14328999999997904524207'
      at Context.eval (http://127.0.0.1:8000/__cypress/tests?p=test/cypress/support/index.js:2073:65)
  Console Logs:
```

### Solution

Uncomment that out and push a PR to verify the tests are still failing because of that line.

Figure out the problem and fix.

**Comment by taoeffect** [comment]
Update: This problem is still happening, because `giAcceptGroupInvite` eventually calls `giCheckIfJoinedChatroom` via `giLogin` => `cy.giCheckIfJoinedGeneralChatroom(username)` => `cy.giCheckIfJoinedChatroom(CHATROOM_GENERAL_NAME, username)`.

And we have this call in `user3 joins the ${groupName1} group and ${CHATROOM_GENERAL_NAME} channel again and logout`:

```js
    cy.giAcceptGroupInvite(invitationLinkAnyone, {
      username: user3,
      groupName: groupName1,
      existingMemberUsername: user1,
      shouldLogoutAfter: false,
      isLoggedIn: true,
      bypassUI: true
    })
```

And in `giCheckIfJoinedChatroom` we have this call:

```js
    if (inviter) {
      cy.get('.c-message:last-child .c-who > span:first-child').should('contain', inviter)
    }
```

Which seems to be where the problem is happening.

**Comment by taoeffect** [comment]
This shows a comparison of a passing (left) and failing (right) build, suggesting that either the last two messages (u2 and u3) are reversed, or the u3 message is missing, or... something even crazier is happening.

![Screenshot 2024-07-26 at 6 56 45 PM](https://github.com/user-attachments/assets/97135f3f-a1ef-40d8-8ef4-4771f4ad9adf)

**Comment by taoeffect** [comment]
Because this heisenbug is such a persistent problem, I've commented out the test as a temporary measure until someone is able to figure it out:

```js
  cy.getByDT('conversationWrapper').within(($el) => {
    if (inviter) {
      // TODO: fix this heisenbug here: https://github.com/okTurtles/group-income/issues/2256
      // cy.get('.c-message:last-child .c-who > span:first-child').scrollIntoView()
      // cy.get('.c-message:last-child .c-who > span:first-child').should('contain', inviter)
    }
```

**Comment by taoeffect** [comment]
It turns out commenting that out in #2280 just leads to a different, related heisenbug!

![Screenshot 2024-08-03 at 11 06 55 AM](https://github.com/user-attachments/assets/e09ac500-8893-47a3-8508-5e57b8a2b45b)

I attempted a fix, which [didn't work](https://github.com/okTurtles/group-income/actions/runs/10230030171/job/28304488155), so now I've pushed a commit to #2280 that gives up completely on performing these checks:

```js
  // failed attempt to fix heisenbug: https://github.com/okTurtles/group-income/issues/2256
  // if (inviter) {
  //   cy.get('[data-test="conversationWrapper"] .c-message:last-child .c-who > span:first-child').should('contain', inviter)
  // }
  // const message = selfJoin ? `Joined ${channelName}` : `Added a member to ${channelName}: ${invitee}`
  // cy.get('[data-test="conversationWrapper"] .c-message:last-child .c-notification').should('contain', message)

  // original code follows:
  // cy.getByDT('conversationWrapper').within(($el) => {
  //   if (inviter) {
  //     // TODO: fix heisenbug: https://github.com/okTurtles/group-income/issues/2256
  //     // cy.get('.c-message:last-child .c-who > span:first-child').scrollIntoView()
  //     // cy.get('.c-message:last-child .c-who > span:first-child').should('contain', inviter)
  //   }
  //   const message = selfJoin ? `Joined ${channelName}` : `Added a member to ${channelName}: ${invitee}`
  //   cy.get('.c-message:last-child .c-notification').should('contain', message)
  // })
```

**Comment by taoeffect** [comment]
Possibly related heisenbug that appears when all of that code is commented out [in this build](https://github.com/okTurtles/group-income/actions/runs/10253330956/job/28365706772):

![Screenshot 2024-08-05 at 10 48 08 AM](https://github.com/user-attachments/assets/a619c65f-c30f-43a8-aeed-5b752aa9f2df)

Here's the video for this one:

https://github.com/user-attachments/assets/49b45c88-afb4-4921-8d8d-25051d4dda31

Which I got by repeatedly running the test locally with:

```
while npx cypress run -c 'baseUrl=http://localhost:8000' --spec "test/cypress/integration/group-chat.spec.js"; do :; done
```

**Comment by taoeffect** [comment]
And yet a different one, also in group-chat, this time *with [video from Cypress Dashboard](https://cloud.cypress.io/projects/q6whky/runs/2890/test-results/8af7d000-7dd9-413c-a9e6-b959450e9498/video?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D&testingTypesEnum=%5B%5D&utm_source=github)*!

![Screenshot 2024-08-05 at 7 03 11 PM](https://github.com/user-attachments/assets/ab01b0f8-5507-4a7f-8309-3d4ba6f7f77d)

**Comment by taoeffect** [comment]
Might've been fixed by @corrideat in #2294
